src/project/mod.rs

```diff
@@ -0,0 +1,126 @@
+use cerberus::project::ProjectData;
+use std::sync::Arc;
+use std::time::Instant;
+
+use cerberus::registry::{RegistryClient, RegistryError, RegistryHttpClient, RegistryResult};
+use common::metrics::AppMetrics;
+use common::storage::{redis, StorageError};
+
+pub use config::*;
+pub use error::*;
+
+use crate::error::{RpcError, RpcResult};
+use crate::project::metrics::ProjectDataMetrics;
+use crate::project::storage::ProjectStorage;
+use crate::project::storage::{Config as StorageConfig, ProjectDataResult};
+
+mod config;
+mod error;
+
+pub mod metrics;
+pub mod storage;
+
+#[derive(Debug, Clone)]
+pub struct Registry {
+    client: RegistryHttpClient,
+    cache: Option<ProjectStorage>,
+    metrics: ProjectDataMetrics,
+}
+
+#[derive(PartialEq, Eq, Debug)]
+pub enum ResponseSource {
+    Cache,
+    Registry,
+}
+
+impl Registry {
+    pub fn new(cfg_registry: &Config, cfg_storage: &StorageConfig) -> RpcResult<Self> {
+        let api_url = &cfg_registry.api_url;
+        let api_auth_token = &cfg_registry.api_auth_token;
+
+        let client = if let (Some(api_url), Some(api_auth_token)) = (api_url, api_auth_token) {
+            RegistryHttpClient::new(api_url, api_auth_token)?
+        } else {
+            return Err(RpcError::InvalidConfiguration(
+                "missing registry api parameters".to_string(),
+            ));
+        };
```

Comment:
There's a very neat new syntax feature [available in 1.65](https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html#let-else-statements):
```suggestion
        let (Some(api_url), Some(api_auth_token)) = (api_url, api_auth_token) else {
            return Err(RpcError::InvalidConfiguration(
                "missing registry api parameters".to_string(),
            ));
        };

        let client = RegistryHttpClient::new(api_url, api_auth_token)?;
```